src/socket/encrypter.rs

```diff
@@ -122,45 +100,52 @@ async fn enc_task(
     client_hs: Option<Handshake>,
     ctx: DaemonContext,
 ) -> anyhow::Result<()> {
-    // get rendezvous ep
-    let rendezvous_ep = match rendezvous_fp {
-        Some(rob) => {
-            // We're the server
-            Endpoint::new(rob, HAVEN_FORWARD_DOCK)
-        }
-        None => {
-            // We're the client: look up Rob's addr in rendezvous dht
-            log::trace!(
-                "alice is about to send an earendil packet! looking up {} in the DHT",
-                remote_ep.fingerprint
-            );
-            let bob_locator = ctx
-                .dht_get(remote_ep.fingerprint)
-                .await
-                .map_err(|_| SocketSendError::DhtError)?
-                .context("Could not get rendezvous for {endpoint}")
-                .map_err(|_| SocketSendError::HavenSendError)?;
-            log::trace!("found rob in the DHT");
-            Endpoint::new(bob_locator.rendezvous_point, HAVEN_FORWARD_DOCK)
-        }
-    };
+    async fn send_to_rendezvous(
```

Comment:
You should be able to make this helper function a closure that captures `ctx`, etc. `let send_to_rendezvous = || async { ...`

This avoids having to pass all the stuff in over and over